src/test/common/clara/test_memory.cljc

```diff
@@ -43,10 +44,48 @@
 
 (use-fixtures :once st/validate-schemas #?(:clj tu/opts-fixture))
 
+(defn join-filter-equals
+  "Intended to be a test function that is the same as equals, but is not visible to Clara as such
+  and thus forces usage of join filters instead of hash joins"
+  [& args]
+  (apply = args))
+
 ;; While the memory is tested through rules and queries, rather than direct unit tests on the memory,
 ;; the intent of these tests is to create patterns in the engine that cover edge cases and other paths
 ;; of concern in clara.rules.memory.
 
+(def-rules-test test-negation-complex-join-with-numerous-non-matching-facts-inserted-after-descendant-negation
```

Comment:
Perhaps a link to the GitHub issue containing more details about what is happening? 
I wouldn't say it is at all obvious what this test is really meant to test. It just looks like a really specific scenario with no rationale given when looking at this later once we've forgotten the initial reason.